### PR TITLE
WIP: Add support to render books on the home page carousel.

### PR DESCRIPTION
**Pull request by Hyperion101010**
Mek and Tabish just if can you leave comment about a sample file that I can
refer to get an idea how the data is rendered on the page.

### Description

It adds carousel to the home page under book sponsor feature.
Closes #

### CODE REVIEW

File: openlibrary/templates/books/custom_carousel.html [hunk 1]
```diff
@@ -1,5 +1,8 @@
 $def with(books=[], title="", url="", key="", min_books=1, load_more=None, test=False)
 
+$def sponsor_me_carousel():
```
Comment (mekarpeles) [inline openlibrary/templates/books/custom_carousel.html]: This will need to be annotated with `@public`

**Comment by Hyperion101010** [comment]
@mekarpeles and @cclauss  I would like to discuss how and what API we are using here.

File: openlibrary/templates/books/custom_carousel.html (hunk 1 above)
Comment (mekarpeles) [inline openlibrary/templates/books/custom_carousel.html]: In order for this method to be accessible to the template, we'd need to use the `@public` decorator :)

**Comment by mekarpeles** [comment]
We have a basic sponsor carousel on the homepage and /sponsorship page. Recommending we close this issue/pr for now. Eventually we'll want to figure out a more sustainable way to let patrons browse sponsorable books (which I think is harder than it seems). Thank you @Hyperion101010 for helping us make progress here!

**Pull request closed without merging**